server_modules/scores.js

```diff
@@ -47,4 +71,46 @@ exports.route = function(app) {
         }).catch(utils.sendError(res)).done();
     });
 
+    //save a new score
+    app.post('/scores/create',function(req,res) {
+        var body = JSON.parse(req.body);
+        var scoresheet = body.scoresheet;
+        var score = body.score;
+
+        changeScores(function(result) {
+            result.scores.push(score);
+            return result;
+        })
+        .then(fileSystem.writeFile(fileSystem.getDataFilePath("scoresheets/" + score.file), req.body))
+        .then(function(scores) {
+            res.json(scores).end();
+        }).catch(utils.sendError(res));
+
+    });
+
+    //delete a score at an id
+    app.post('/scores/delete/:id',function(req,res) {
+        changeScores(function(result) {
+            var index = result.scores.findIndex((score) => score.id === req.params.id);
+            result.scores.splice(index, 1);
+            return result;
+        }).then(function(scores) {
+            res.json(scores).end();
+        }).catch(utils.sendError(res)).done();
+    });
+
+    //edit a score at an id
+    app.post('/scores/update/:id',function(req,res) {
+        var score = JSON.parse(req.body);
+        changeScores(function(result) {
+            var index = result.scores.findIndex((score) => score.id === req.params.id);
+            result.scores.splice(index, 1);
+            result.scores.push(score);
```

Comment:
The scores view shows entries in chronological order of insertion, so moving it to the end is undesirable.
Why not simply `result.scores[index] = score`?